api/candlepin-api-spec.yaml

```diff
@@ -30,6 +30,336 @@ paths:
         default:
           $ref: '#/components/responses/default'
 
+  /activation_keys:
+    get:
+      description: Retrieves a list of activation keys
+      tags:
+        - Activation Key
+      operationId: findActivationKey
+      x-java-response:
+        type: Iterable
+        isContainer: true
+      security: []
+      responses:
+        200:
+          description: List of activation keys
+          content:
+            application/json:
+              schema:
+                type: array
+                items:
+                  $ref: '#/components/schemas/ActivationKeyDTO'
+        default:
+          $ref: '#/components/responses/default'
+
+  /activation_keys/{activation_key_id}:
+    get:
+      description: Retrieves a single activation key
+      tags:
+        - Activation Key
+      operationId: getActivationKey
+      security: []
+      parameters:
+        - name: activation_key_id
+          in: path
+          description: The Id of activation key to retrieve
+          required: true
+          schema:
+            type: string
+      responses:
+        200:
+          description: Activation key
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ActivationKeyDTO'
+        400:
+          description: Activation key Id is null or empty or could not be found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ExceptionMessage'
+              example:
+                displayMessage: Activation key Id is null or empty or could not be found
+                requestUuid: c4347004-8792-41fe-a4d8-fccaa0d3898a
+        default:
+          $ref: '#/components/responses/default'
+
+    put:
+      description: Updates an Activation Key
+      tags:
+        - Activation Key
+      operationId: updateActivationKey
+      security: []
+      parameters:
+        - name: activation_key_id
+          in: path
+          description: The Id of activation key to be updated
+          required: true
+          schema:
+            type: string
+      requestBody:
+        description: Activation key to be updated
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/ActivationKeyDTO'
+      responses:
+        200:
+          description: Updated activation key
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ActivationKeyDTO'
+        400:
+          description: Validation failure, Null or Empty or Unknown Activation key Id. Either service level
```

Comment:
Lots of unnecessary capitalization here: "Null," "Empty," "Unknown," and "Activation key" are not proper nouns and should not be capitalized. The service level/activation key name validation failure at the end is also a bit confusing, as it seems like it should be a separate message entirely with the way it's written.

I'm assuming this is intending to collect information about every possible validation failure on the key as a whole, but that's likely going to date itself eventually. We should probably just focus on expressing that the validation failed somehow, and leave the details to the exception.

Reply:
Done